src/sentry/receivers/email.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import absolute_import
+
+from django.db import IntegrityError, transaction
+from django.db.models.signals import post_delete, post_save
+
+from sentry.models import Email, UserEmail
+
+
+def create_email(instance, created, **kwargs):
+    if created:
+        try:
+            with transaction.atomic():
+                Email.objects.create(email=instance.email)
+        except IntegrityError:
+            pass
+
+
+def delete_email(instance, **kwargs):
+    if UserEmail.objects.filter(email=instance.email).exists():
```

Comment:
I want to note here, that this might be slightly inaccurate due to this not being a case insensitive lookup. You should do `email__iexact`. Because `Email` is case insensitive but `UserEmail` is, you'll get false positives.

Reply:
Good point: https://github.com/getsentry/sentry/pull/6070